spec/controllers/questions_controller_spec.rb

```diff
@@ -0,0 +1,129 @@
+require 'rails_helper'
+
+describe QuestionsController do
+  let(:question) { create(:question) }
+
+  describe 'GET #index' do
+    let(:questions) { create_list(:question, 2) }
+
+    before { get :index }
+
+    it 'returns array of all questions' do
+      expect(assigns(:questions)).to match_array(questions)
+    end
+
+    it 'renders index template' do
+      expect(response).to render_template :index
+    end
+  end
+
+  describe 'GET #show' do
+    before { get :show, id: question }
+
+    it 'returns one question object' do
+      expect(assigns(:question)).to eq question
+    end
+
+    it 'renders show template' do
+      expect(response).to render_template :show, id: question
+    end
+  end
+
+  describe 'GET #new' do
+    before { get :new }
+
+    it 'assigns new Question object to @question' do
+      expect(assigns(:question)).to be_a_new Question
+    end
+
+    it 'renders new template' do
+      expect(response).to render_template :new
+    end
+  end
+
+  describe 'GET #edit' do
+    before { get :edit, id: question }
+
+    it 'assigns requested question to @question' do
+      expect(assigns(:question)).to eq question
+    end
+
+    it 'renders edit template' do
+      expect(response).to render_template :edit, id: question
+    end
+  end
+
+  describe 'POST #create' do
+    context 'with valid attributes' do
+      it 'saves new question in database' do
+        expect { post :create, question: attributes_for(:question) }.to change(Question, :count).by 1
+      end
+
+      it 'redirects to created question view' do
+        post :create, question: attributes_for(:question)
+        expect(response).to redirect_to question_path(assigns(:question))
+      end
+    end
+
+    context 'with invalid attributes' do
+      it 'doesn\'t save new question' do
+        expect { post :create, question: attributes_for(:invalid_question) }.to_not change(Question, :count)
+      end
+
+      it 'render new template again' do
+        post :create, question: attributes_for(:invalid_question)
+        expect(response).to render_template :new
+      end
+    end
+  end
+
+  describe 'PATCH #update' do
+    context 'valid attributes' do
+      it 'assigns requested question record to @question' do
+        patch :update, id: question, question: attributes_for(:question)
+        expect(assigns(:question)).to eq question
+      end
+
+      it 'updates @question attributes' do
+        patch :update, id: question, question: { title: 'Brand new title', body: 'Shiny body' }
```

Comment:
Line is too long. [95/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.